cluster-autoscaler/core/static_autoscaler_test.go

```diff
@@ -0,0 +1,238 @@
+			ScaleDownUtilizationThreshold: 0.5,
+			MaxNodesTotal:                 1,
+		},
+		PredicateChecker:     simulator.NewTestPredicateChecker(),
+		CloudProvider:        provider,
+		ClientSet:            fakeClient,
+		Recorder:             fakeRecorder,
+		ExpanderStrategy:     random.NewStrategy(),
+		ClusterStateRegistry: clusterState,
+		LogRecorder:          fakeLogRecorder,
+	}
+
+	listerRegistry := kube_util.NewListerRegistry(allNodeListerMock, readyNodeListerMock, scheduledPodMock,
+		unschedulablePodMock, podDisruptionBudgetListerMock, daemonSetListerMock)
+
+	sd := NewScaleDown(context)
+
+	autoscaler := &StaticAutoscaler{AutoscalingContext: context,
+		ListerRegistry:           listerRegistry,
+		lastScaleUpTime:          time.Now(),
+		lastScaleDownFailedTrial: time.Now(),
+		scaleDown:                sd}
+
+	// MaxNodesTotal reached.
+	readyNodeListerMock.On("List").Return([]*apiv1.Node{n1}, nil).Once()
+	allNodeListerMock.On("List").Return([]*apiv1.Node{n1}, nil).Once()
+	scheduledPodMock.On("List").Return([]*apiv1.Pod{p1}, nil).Once()
+	unschedulablePodMock.On("List").Return([]*apiv1.Pod{p2}, nil).Once()
+	podDisruptionBudgetListerMock.On("List").Return([]*policyv1.PodDisruptionBudget{}, nil).Once()
+
+	err := autoscaler.RunOnce(time.Now())
+	assert.NoError(t, err)
+	mock.AssertExpectationsForObjects(t, readyNodeListerMock, allNodeListerMock, scheduledPodMock, unschedulablePodMock,
+		podDisruptionBudgetListerMock, daemonSetListerMock, onScaleUpMock, onScaleDownMock)
+
+	// Scale up.
+	readyNodeListerMock.On("List").Return([]*apiv1.Node{n1}, nil).Once()
+	allNodeListerMock.On("List").Return([]*apiv1.Node{n1}, nil).Once()
+	scheduledPodMock.On("List").Return([]*apiv1.Pod{p1}, nil).Once()
+	unschedulablePodMock.On("List").Return([]*apiv1.Pod{p2}, nil).Once()
+	daemonSetListerMock.On("List").Return([]*extensionsv1.DaemonSet{}, nil).Once()
+	onScaleUpMock.On("ScaleUp", "ng1", 1).Return(nil).Once()
+
+	context.MaxNodesTotal = 10
+	err = autoscaler.RunOnce(time.Now())
+	assert.NoError(t, err)
+	mock.AssertExpectationsForObjects(t, readyNodeListerMock, allNodeListerMock, scheduledPodMock, unschedulablePodMock,
+		podDisruptionBudgetListerMock, daemonSetListerMock, onScaleUpMock, onScaleDownMock)
+
+	// Do nothing.
+	readyNodeListerMock.On("List").Return([]*apiv1.Node{n1, n2}, nil).Once()
+	allNodeListerMock.On("List").Return([]*apiv1.Node{n1, n2}, nil).Once()
+	scheduledPodMock.On("List").Return([]*apiv1.Pod{p1}, nil).Once()
+	unschedulablePodMock.On("List").Return([]*apiv1.Pod{p2}, nil).Once()
+	podDisruptionBudgetListerMock.On("List").Return([]*policyv1.PodDisruptionBudget{}, nil).Once()
+
+	provider.AddNode("ng1", n2)
+
+	err = autoscaler.RunOnce(time.Now())
+	assert.NoError(t, err)
+	mock.AssertExpectationsForObjects(t, readyNodeListerMock, allNodeListerMock, scheduledPodMock, unschedulablePodMock,
+		podDisruptionBudgetListerMock, daemonSetListerMock, onScaleUpMock, onScaleDownMock)
+
+	// Scale down.
```

Comment:
Please add a comment (or snippet) why scale down is possible just after scale up.

Reply:
Done

Reply:
Changed duration to one minute.

Reply:
Not sure what does it mean exactly but we will not put 1 minute sleep in tests :).

Reply:
Changed ScaleDownUnreadyTime and ScaleDownUnneededTime  to one minute.